# Null hypothesis: 
# The implementation of an alternative bus route for bus line X8 increases or does not change the duration of bus trips.
## _$H_0$_:  $t_{\mathrm{1}}- t_{\mathrm{0}} > = 0$ 
# Alternative hypothesis:
# The implementation of an alternative bus route for bus line X8 decreases the duration of bus trips.
## _$H_1$_:  $t_{\mathrm{1}}- t_{\mathrm{0}} < 0$ 
## significance level  $\alpha=0.05$

from https://github.com/fedhere/PUI2018_fb55/blob/master/Lab4_fb55/times.txt save and upload raw data times.txt to local directory.

In [1]:
import pandas as pd
import numpy as np
from math import sqrt

In [11]:
times = pd.read_csv("times.txt", header = None)
times.head()

0
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

Given 36 +- 6 minutes (mean and standard deviation parameters of the population), we get mu of population = 36, sigma of population = 6. 

We plug these values into the formula:

In [12]:
z = (36 - times.mean())/(6/sqrt(len(times)))
print("z score: {}". format(z[0]))

z score: 2.55639718617


So the number of standard deviations away from the mean of my sample (times) is 2.56, meaning that we are farther than 2 sigmas away from the sample mean.

Because we took a significance level of alpha = 0.05, which does not reject the null hypothesis if the sigma falls inside 2 standard deviations (95%) of our sample, we reject the null hypothesis that the implementation of an alternative bus route for bus line X8 increases or does not change the duration of bus trips.

This means that our conclusion is that the implementation of an alternative bus route for bus line X8 decreases the duration of bus trips (alternative hypothesis).